admiral/pkg/controller/common/types.go

```diff
@@ -186,6 +189,18 @@ func (s *SidecarEgressMap) Delete(key string) {
 	delete(s.cache, key)
 }
 
+// Map func returns a map of identity to namespace:SidecarEgress map
+// Iterating through the returned map is not implicitly thread safe,
+// use (s *SidecarEgressMap) Range() func instead.
 func (s *SidecarEgressMap) Map() map[string]map[string]SidecarEgress {
 	return s.cache
 }
+
+// Range is a thread safe iterator to iterate through the SidecarEgress map
+func (s *SidecarEgressMap) Range(fn func(k string, v map[string]SidecarEgress)) {
```

Comment:
Where will this be used?

Reply:
hey @nirvanagit , I've added this func to provide a thread-safe way to iterate through the `SidecarEgress` map. Basically discouraging using the `.Map()` func to get access to the underlying map and then performing a `for range` on it. I've added comments to make it clear.

Reply:
Is this already being used ?

Reply:
No it's not being used right now